crt_portal/cts_forms/forms.py

```diff
@@ -1262,12 +1262,155 @@ class PrintActions(Form):
+                'class': 'usa-textarea',
+            },
+        ),
+    )
+
+    def get_updates(self):
+        return {field: self.cleaned_data[field] for field in self.changed_data}
+
+    def get_update_description(self):
+        """
+        Given a submitted form, emit a textual description of what was updated.
+        """
+        labels = {key: self.fields[key].label or key for key in self.changed_data}
+        labels.pop('comment', None)  # required, so we can omit
+        default_string = '{what} set to {item}'
+        custom_strings = {
+            'assigned to': 'assigned to {item}',
+            'crt summary': 'summary updated',
+        }
+        descriptions = []
+        for (key, value) in labels.items():
+            what = value.lower()
+            item = self.cleaned_data[key]
+            string = custom_strings.get(what, default_string)
+            description = string.format(**{'what': what, 'item': item})
+            descriptions.append(description)
+        if len(descriptions) > 1:
+            descriptions[-1] = f'and {descriptions[-1]}'
+        return ', '.join(descriptions) or 'comment added'
+
+    def get_actions(self, report):
+        """
+        Parse incoming changed data for activity stream entry (tweaked for
+        bulk update)
+        """
+        for field in self.changed_data:
+            name = ' '.join(field.split('_')).capitalize()
+            # rename primary statute if applicable
+            if field == 'primary_statute':
+                name = 'Primary classification'
+            if field in ['summary', 'comment']:
+                continue
+            initial = getattr(report, field, 'None')
+            yield f"{name}:", f'Updated from "{initial}" to "{self.cleaned_data[field]}"'
+
+    def update_activity_stream(self, user, report):
+        """
+        Send all actions to activity stream (tweaked for bulk update)
+        """
+        for verb, description in self.get_actions(report):
+            add_activity(user, verb, description, report)
+
+    def update(self, reports, user):
+        """
+        Bulk update given reports and update activity log for each report
+        """
+        updated_data = self.get_updates()
+        comment_string = updated_data.pop('comment', None)
+        summary_string = updated_data.pop('summary', None)
+
+        # update activity log _before_ we update fields so
+        # that we still have access to the original field
+        for report in reports:
+            self.update_activity_stream(user, report)
+
+        if comment_string:
+            kwargs = {
+                'is_summary': False,
+                'note': comment_string,
+                'author': user.username,
+            }
+            comment = CommentAndSummary.objects.create(**kwargs)
+            for report in reports:
+                report.internal_comments.add(comment)
+                add_activity(user, 'Added comment: ', comment_string, report)
```

Comment:
This jumps out since we're creating a single `CommentAndSummary` instance and then associating it with _all_ of the modified reports.  Comment's are not editable so we won't encounter the bug you hit for Summaries but this feels like we may run into problems/confusion in the future.

A broader discussion, should we move towards separating `CommentAndSummary` into two separate models which are ForeignKeys to `Report`?

Reply:
Right. If we don't use this `internal_comments` field as intended (that is, as m2m) then what's the point?

I'm happy to create one for each report here if that makes things simpler down the road. Untangling the m2m relationship here into two FKs may not be too difficult, but admittedly I haven't looked at this in depth yet. Let's make a separate ticket? Meanwhile I'll update this PR to do one comment per report.

Reply:
(Updated to create one comment per report)

Reply:
Agreed, we'll add a new card to:

Consider and make a decision on breaking down `CommendAndSummary`